pypesto/hierarchical/calculator.py

```diff
@@ -159,3 +175,31 @@ def __call__(
         result[INNER_RDATAS] = inner_rdatas
 
         return result
+
+    def set_simulation_edatas(
+        self,
+        edatas,
+        simulation_timepoints: Sequence[Sequence[Union[float, int]]] = None,
+    ):
+        """Set the experimental data used for simulation.
+
+        This is required to simulate model trajectories at more timepoints than the
+        measurement timepoints. Checks ensure that the experimental data used for
+        simulation is a subset of the experimental data used for the inner problem.
+
+        Parameters
+        ----------
+        simulation_timepoints:
+            The outer sequence should contain a sequence of timepoints for each
+            experimental condition.
+        """
+
+        if simulation_timepoints is None:
+            self.simulation_edatas = None
+            return
+        else:
+            simulation_edatas = copy.deepcopy(edatas)
+            for i, edata in enumerate(simulation_edatas):
+                edata.setTimepoints(simulation_timepoints[i])
```

Comment:
Note that just changing the timepoints will still keep the values of the measurements/stddevs which will then be associated with the wrong timepoints.